### DISCUSSION TITLE
How to export a component group with styles included

### DISCUSSION THREAD

**Discussion by edwinjoel1984** [GrapesJS/grapesjs#3799] [Q&A]
I want to save a group of components (components with children) in the database to later use them as drawable components within the editor, but I have not been able to save the component with their styles.

I have tried using `editor.getSelected().ToJSON()`  but this only brings me the components without styles and for the styles I have tried with `editor.CodeManager.getCode (yourComponent, 'css', {cssc: editor.CssComposer})` , and I have not got any results.

Any ideas on how to achieve this?

**Comment by edwinjoel1984** [comment]
@artf  Could you please help?

**Comment by artf** [comment]
From [this version](https://github.com/artf/grapesjs/releases/tag/v0.17.22) you should be able to export CSS rules by component.
```js
editor.CodeManager.getCode(component, 'css', { json: true })
```

**Reply by edwinjoel1984** [reply]
Excelent, but how can I add those styles back to the editor?

**Reply by artf** [reply]
Try to use [addRules](https://grapesjs.com/docs/api/css_composer.html#addrules), I know in docs it tells the argument should be a CSS string, but it should also work with a JSON
